File: backend/src/main/java/dev/coldhands/pair/stairs/backend/usecase/UserDisplayNameService.java
package dev.coldhands.pair.stairs.backend.usecase;

import dev.coldhands.pair.stairs.backend.domain.UserName;

public class UserDisplayNameService {

    public String getDisplayNameFor(UserName userName) {

        if (userName.nickName() != null) {
            return userName.nickName();
        }

        if (userName.givenName() != null) {
            return userName.givenName();
        }

        if (userName.fullName() != null) {
            final var fullName = userName.fullName();

            return fullName.split(" ", 2)[0];
        }

        return "Unknown";
    }
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/usecase/UserDisplayNameService.kt
package dev.coldhands.pair.stairs.backend.usecase

import dev.coldhands.pair.stairs.backend.domain.UserName

class UserDisplayNameService {

    fun getDisplayNameFor(userName: UserName): String =
        userName.nickName
            ?: userName.givenName
            ?: userName.fullName?.let { it.split(" ", limit = 2)[0] }
            ?: "Unknown"
}